djangocms_googlemap/models.py

```diff
@@ -1,90 +1,287 @@
+"""
+Enables the user to add various google maps plugin that adds maps or images
+from the google maps api.
+"""
+from __future__ import unicode_literals
+
+import re
+import json
+
 from django.db import models
-from django.utils.translation import ugettext_lazy as _
+from django.conf import settings
+from django.core.exceptions import ValidationError
+from django.utils.encoding import python_2_unicode_compatible
+from django.utils.translation import ugettext, ugettext_lazy as _
 
 from cms.models import CMSPlugin
-from cms.utils.compat.dj import python_2_unicode_compatible
 
 
-@python_2_unicode_compatible
-class GoogleMap(CMSPlugin):
-    """
-    A google maps integration
-    """
-    translatable_content_excluded_fields = [
-        'address', 'zipcode', 'width', 'height']
+MAP_TYPES = ['ROADMAP', 'SATELLITE', 'HYBRID', 'TERRAIN']
+MAP_TYPE_CHOICES = [(map_type, map_type) for map_type in MAP_TYPES]
+TRAVEL_MODES = ['DRIVING', 'BICYCLING', 'TRANSIT', 'WALKING']
+TRAVEL_MODE_CHOICES = [(travel_mode, travel_mode) for travel_mode in TRAVEL_MODES]
+ZOOM_LEVEL_CHOICES = [(c, str(c)) for c in range(22)] # from 0 to 21
 
-    title = models.CharField(_("map title"), max_length=100, blank=True,
-                             null=True)
+CSS_WIDTH_RE = re.compile(r'^\d+(?:px|%)$')
+CSS_HEIGHT_RE = re.compile(r'^\d+px$')
 
-    address = models.CharField(_("address"), max_length=150)
-    zipcode = models.CharField(_("zip code"), max_length=30)
-    city = models.CharField(_("city"), max_length=100)
 
-    content = models.CharField(
-        _("additional content"), max_length=255, blank=True,
-        help_text=_('Displayed under address in the bubble.'))
+# Add additional choices through the ``settings.py``.
+def get_templates():
+    choices = [
+        ('default', _('Default')),
+    ]
+    choices += getattr(
+        settings,
+        'DJANGOCMS_GOOGLEMAP_TEMPLATES',
+        [],
+    )
+    return choices
 
-    style = models.TextField(_("custom map style"), blank=True,
-        help_text=_('Provide a valid JSON configuration (escaped). See '
-            'developers.google.com/maps/documentation/javascript/styling'))
 
-    ZOOM_LEVELS = map(lambda c: (c, str(c)), range(22))
+@python_2_unicode_compatible
+class GoogleMap(CMSPlugin):
+    """
+    Renders the Google Maps wrapper
+    """
+    template = models.CharField(
+        verbose_name=_('Template'),
+        choices=get_templates(),
+        default=get_templates()[0][0],
+        max_length=255,
+    )
+    title = models.CharField(
+        verbose_name=_('Map title'),
+        max_length=255,
+        blank=True,
+    )
+    width = models.CharField(
+        verbose_name=_('Width'),
+        max_length=6,
+        default='100%',
+        help_text=_('Width of the map (e.g. "100%" or "400px").'),
+    )
+    height = models.CharField(
+        verbose_name=_('Height'),
+        max_length=6,
+        default='400px',
+        help_text=_('Height of the map (e.g. "400px").'),
```

Comment:
``'Height of the map in pixels (e.g. "400").'``

I would keep it as a charfield and clean it, removing the ``px`` if the user has added it.